src/low-level/smtp/mailsmtp.c

```diff
@@ -448,6 +448,65 @@ int mailsmtp_data_message(mailsmtp * session,
   }
 }
 
+int mailsmtp_status(int smtpstatus)
+{
+  switch(smtpstatus) {
+  case 250:
+    return MAILSMTP_NO_ERROR;
+
+  case 552:
+    return MAILSMTP_ERROR_EXCEED_STORAGE_ALLOCATION;
+
+  case 554:
+    return MAILSMTP_ERROR_TRANSACTION_FAILED;
+
+  case 451:
+    return MAILSMTP_ERROR_IN_PROCESSING;
+
+  case 452:
+    return MAILSMTP_ERROR_INSUFFICIENT_SYSTEM_STORAGE;
+
+  case 0:
+    return MAILSMTP_ERROR_STREAM;
+
+  default:
+    return MAILSMTP_ERROR_UNEXPECTED_CODE;
+  }
+
+}
+
+/* lmtp data command
+ * on the end of data transaction lmtp returns a status for
+ * every recipient it delivered.
+ * recipient_list holds the rcpts from smtp_rcpt()
+ * the last error is returned to the caller
+ * if retcodes is not NULL, the individual rcpt-stati are stored there
+ *  */
+int maillmtp_data_message(mailsmtp * session,
+			   const char * message,
+			   size_t size,
+               clist * recipient_list,
+               clist * retcodes)
```

Comment:
btw, clist usually contain pointers. It's unsafe to store integers in the clist.
You need to allocate pointers and add them to the list.